dbscripts/xml/upgrade/3.2.0_preupdate_versioning.xml

```diff
@@ -47,4 +47,13 @@
 		<dropindex table="citations" index="citations_submission_seq" />
 		<rename table="citations" column="submission_id" to="publication_id" />
 	</sql>
+
+	<!--
+	Necessary to allow longer supplementary files titles to be added as a label when
+	they are converted to public galleys. This happens with previous upgrades from 2.4.x
+	to other versions of OJS 3, but does not otherwise happen with the 2.4.x to 3.2.x upgrade.
+ -->
+	<sql>
+		<query>ALTER TABLE temp_submission_galleys MODIFY COLUMN label VARCHAR(255);</query>
```

Comment:
The `label` column will already be varchar(255) for OJS 3.x installations, but it'll be varchar(32) for OJS 2.x installations.

I think this needs two changes:
1. Move it to `dbscripts/xml/upgrade/3.0.0_preupdate.xml`, just after `article_galleys` is renamed to `submission_galleys`, so it's only applied when needed
2. I don't think this SQL will work for PostgreSQL, so you'd need to add a clause for each DBMS and adapt.